src/main/java/org/spongepowered/api/util/blockray/BlockRay.java

```diff
@@ -0,0 +1,650 @@
+                this.filter = filter;
+            } else {
+                this.filter = Predicates.and(this.filter, filter);
+            }
+            return this;
+        }
+
+        /**
+         * Sets the direction and ending location. This or setting the direction is required and can only be done once.
+         *
+         * @param end The ending location
+         * @return This for chained calls
+         */
+        public BlockRayBuilder to(Vector3d end) {
+            Preconditions.checkArgument(this.direction == null, "Direction has already been set");
+            Preconditions.checkArgument(end != null, "End cannot be null");
+            Preconditions.checkArgument(!this.position.equals(end), "Start and end cannot be equal");
+            this.direction = end.sub(this.position).normalize();
+            return filter(new TargetBlockFilter(end));
+        }
+
+        /**
+         * Sets the direction. This or setting the ending location is required and can only be done once.
+         *
+         * @param direction The direction
+         * @return This for chained calls
+         */
+        public BlockRayBuilder direction(Vector3d direction) {
+            Preconditions.checkArgument(this.direction == null, "Direction has already been set");
+            Preconditions.checkArgument(direction != null, "Direction cannot be null");
+            Preconditions.checkArgument(direction.lengthSquared() != 0, "Direction must be a non-zero vector");
+            this.direction = direction.normalize();
+            return this;
+        }
+
+        /**
+         * Sets the maximum number of blocks to intersect before stopping.
+         * This is a safeguard to prevent infinite iteration.
+         * Default value is 1000. Use a negative value to disable this.
+         *
+         * @param blockLimit The block limit
+         * @return This for chained calls
+         */
+        public BlockRayBuilder blockLimit(int blockLimit) {
+            this.blockLimit = blockLimit;
+            return this;
+        }
+
+        /**
+         * Returns a block ray build from the settings. An ending location or direction needs to have been set.
+         *
+         * @return A block ray
+         */
+        public BlockRay build() {
+            Preconditions.checkState(this.direction != null, "Either end point or direction needs to be set");
+            final BlockRay blockRay = new BlockRay(this.filter, this.extent, this.position, this.direction);
+            blockRay.setBlockLimit(this.blockLimit);
+            return blockRay;
+        }
+
+        @Override
+        public Iterator<BlockRayHit> iterator() {
+            return build();
+        }
+    }
+
+    /**
+     * A filter that only allows blocks of a certain block type.
+     *
+     * @param type The type of blocks to allow
+     * @return The filter instance
+     */
+    public static Predicate<BlockRayHit> blockTypeFilter(final BlockType type) {
```

Comment:
There's only a few of them, I don't think it's worth it. Adding a "filter" suffix makes it clear enough.

> On Jun 17, 2015, at 4:38 PM, gratimax <notifications@github.com> wrote:
> 
> In src/main/java/org/spongepowered/api/util/blockray/BlockRay.java:
> 
> > +            return blockRay;
> > +        }
> > +
> > +        @Override
> > +        public Iterator<BlockRayHit> iterator() {
> > +            return build();
> > +        }
> > +    }
> > +
> > +    /**
> > +     * A filter that only allows blocks of a certain block type.
> > +     *
> > +     * @param type The type of blocks to allow
> > +     * @return The filter instance
> > +     */
> > +    public static Predicate<BlockRayHit> blockTypeFilter(final BlockType type) {
> I feel like methods like these, related to block ray filters belong on another class, since they aren't strictly related to dealing with block rays themselves.
> 
> —
> Reply to this email directly or view it on GitHub.
>